tests/datasets/entities/test_entity_key.py

```diff
@@ -0,0 +1,14 @@
+import pytest
+
+from snuba.datasets.entities.entity_key import REGISTERED_ENTITY_KEYS, EntityKey
+from snuba.datasets.entities.factory import reset_entity_factory
+
+
+def test_entity_key() -> None:
+    reset_entity_factory()
```

Comment:
Does this function only exist for testing? I can't imagine any other reason why it would need to.

If you need a blank entity factory for a test, just mock the `_ent_factory` function to return what you want

https://github.com/getsentry/snuba/blob/53fadadab46b75aa0bf456c5e1d256eeb962fca3/snuba/datasets/entities/factory.py#L120

Reply:
this reset method shouldn't exist because it invites mutability into global state, which we should not have.

Reply:
I'll remove it, this was introduced as part of #3065 and I just used it here

Reply:
This was introduced purely for testing the multithreaded race condition. As opposed to spawning new threads and accessing the entity factory in a test case, the test was revised to ensure the entity factory initialization and the entity mapping attribute creation are not decoupled.

Reply:
(off-topic for this PR)

It's best to keep testing functions within tests especially for things like this which work with global state. Having a function that alters global state, should not be used in normal operation of the system, in the public api of a module introduces more surface area for issues.

This is a usecase for the python mocking library which can inject these kinds of corner conditions into the runtime of the test but without changing the way the system operates